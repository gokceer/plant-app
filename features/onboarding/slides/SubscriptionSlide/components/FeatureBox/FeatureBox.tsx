import React from "react";
import {
  Image,
  ImageSourcePropType,
  Platform,
  StyleSheet,
  Text,
  View,
} from "react-native";

const FeatureBox = ({
  icon,
  title,
  subtitle,
}: {
  icon: ImageSourcePropType;
  title: string;
  subtitle: string;
}) => (
  <View style={styles.featureBox}>
    <Image source={icon} style={styles.icon} />
    <Text style={styles.featureTitle}>{title}</Text>
    <Text style={styles.featureSubtitle}>{subtitle}</Text>
  </View>
);

export default FeatureBox;

const styles = StyleSheet.create({
  featureBox: {
    backgroundColor: "#FFFFFF14",
    padding: 16,
    borderRadius: 12,
    width: 156,
    height: 130,
    marginRight: 8,
  },
  icon: {
    width: 36,
    height: 36,
  },
  featureTitle: {
    color: "#fff",
    fontFamily: "RubikMedium",
    fontSize: 20,
    paddingTop: 16,
  },
  featureSubtitle: {
    color: "#FFFFFFB2",
    fontFamily: "RubikRegular",
    fontSize: 12,
    ...Platform.select({ ios: { paddingTop: 4 } }),
  },
});
