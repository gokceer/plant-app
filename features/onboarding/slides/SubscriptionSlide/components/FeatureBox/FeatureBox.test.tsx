import { render } from "@testing-library/react-native";
import React from "react";
import FeatureBox from "./FeatureBox";

describe("FeatureBox", () => {
  const props = {
    icon: { uri: "https://example.com/image1.jpg" },
    title: "Unlimited",
    subtitle: "Plant Identify",
  };

  it("renders correctly and matches snapshot", () => {
    const { toJSON } = render(<FeatureBox {...props} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
