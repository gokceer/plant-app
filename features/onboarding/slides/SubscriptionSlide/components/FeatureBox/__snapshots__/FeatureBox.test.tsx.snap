// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeatureBox renders correctly and matches snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFFFFF14",
      "borderRadius": 12,
      "height": 130,
      "marginRight": 8,
      "padding": 16,
      "width": 156,
    }
  }
>
  <Image
    source={
      {
        "uri": "https://example.com/image1.jpg",
      }
    }
    style={
      {
        "height": 36,
        "width": 36,
      }
    }
  />
  <Text
    style={
      {
        "color": "#fff",
        "fontFamily": "RubikMedium",
        "fontSize": 20,
        "paddingTop": 16,
      }
    }
  >
    Unlimited
  </Text>
  <Text
    style={
      {
        "color": "#FFFFFFB2",
        "fontFamily": "RubikRegular",
        "fontSize": 12,
        "paddingTop": 4,
      }
    }
  >
    Plant Identify
  </Text>
</View>
`;
